doc/plan-for-new-modules-implementation.md

```diff
@@ -0,0 +1,100 @@
+
+At every phase, the following standards must be maintained:
+
+* Spec compliance ([#132](https://github.com/nodejs/modules/issues/132)): We must always follow the ES spec.
+* Browser equivalence ([#133](https://github.com/nodejs/modules/issues/133)): There’s room for debate in specific cases, but in general if Node is doing something that browsers also do, Node should do it in the same way. Alternatively, code that executes in both environments should produce identical results.
+* Don’t break CommonJS ([#112](https://github.com/nodejs/modules/issues/112)): We cannot cause breaking changes with regards to CommonJS.
+
+See also the [features list in the README](https://github.com/nodejs/modules#features).
+
+## Phase 1: The Minimal Kernel
+
+These features will be part of the first phase of development:
+
+* `createRequireFunction` ([nodejs/node#19360](https://github.com/nodejs/node/pull/19360)) is the only way to import CommonJS into an ES module, for now.
+  - `import.meta.require` does not fail early enough.
+  - Hold off on `import` statements for CommonJS until more progress is made on the dynamic modules spec.
+
+* `import` statements will only support files with an `.mjs` extension, and will import only ES modules, for now.
+  - In a later phase, the intention is to move forward with format databases to map extensions and support multiple use cases.
+  - No JSON or native modules; `createRequireFunction` can be used to get these.
+
+
+### How will we get from where we are to Phase 1
+
+* Remove support in the `import` statement of formats other than ESM:
+  - No CommonJS.
+  - No JSON.
+  - No native modules.
+  - Implemented in https://github.com/nodejs/ecmascript-modules/pull/3
+
+* Remove dynamic path searching:
+  - No extension adding.
+  - No directory resolution, including no support for `index.js` or `index.mjs`.
+  - Still support the `main` field, but it must specify a full filename.
+  - Implemented in https://github.com/nodejs/ecmascript-modules/pull/2
+
+* Add `createRequireFunction`.
+  - Implemented in https://github.com/nodejs/node/pull/19360
+
+
+## Features Expected in Later Phases
+
+These features are agreed upon, but make sense to include in a later phase of development; or there is agreement on the need for this feature, but a lack of consensus on the implementation details:
+
+* Browser-compatible specifier resolution ([#109](https://github.com/nodejs/modules/issues/109)), a.k.a. bare imports.
```

Comment:
This is in conflict with supporting the `main` field above.

we either support bare imports or we don't

bare imports with a `main` resolution as stated above is the equivelent to support bare imports that are 100% compatible with package name maps

Reply:
Okay, so move the reference to `main` field into this item? You’re right, if we don’t have bare imports in phase 1 then we wouldn’t have `main` there either (for ESM, anyway).